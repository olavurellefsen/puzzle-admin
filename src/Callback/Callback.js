import React from "react";
import { CallbackContainer, LoadingImage } from './Callback.style'; 
import loading from "./loading.svg";

export default () => {
  return (
    <CallbackContainer>
      <LoadingImage src={loading} alt="loading" data-testid="LoadingImage" />
    </CallbackContainer>
  );
}
