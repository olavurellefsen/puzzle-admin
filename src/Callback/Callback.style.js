import styled from 'styled-components';

export const CallbackContainer = styled.div`
`;

export const LoadingImage = styled.img`
`;