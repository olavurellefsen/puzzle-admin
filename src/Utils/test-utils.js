import React from "react";
import { render } from "@testing-library/react";
import { ApolloProvider } from "react-apollo";
import MainContextStore from "../MainContextStore";
import client from "../heyBreydLocalizationClient";

const AllTheProviders = ({ children }) => {
  return (
    <ApolloProvider client={client}>
      <MainContextStore>{children}</MainContextStore>
    </ApolloProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
