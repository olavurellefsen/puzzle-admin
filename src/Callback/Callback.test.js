import React from "react";
import { render, cleanup } from 'test-utils';
import Callback from "./Callback";
afterEach(cleanup)

it("should render correctly", () => {
  const { container } = render(<Callback />);
  expect(container).toMatchSnapshot();
});

it("should show the Loading image", () => {
  const { queryByTestId } = render(<Callback />);
  expect(queryByTestId("LoadingImage")).toBeTruthy();
});