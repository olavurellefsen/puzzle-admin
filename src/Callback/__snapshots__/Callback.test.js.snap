// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa fePDys"
  >
    <img
      alt="loading"
      class="sc-bwzfXH bFHdpU"
      data-testid="LoadingImage"
      src="loading.svg"
    />
  </div>
</div>
`;
